pkg/apis/eventing/v1alpha1/eventpolicy_validation.go

```diff
@@ -20,10 +20,18 @@ import (
 	"context"
 	"strings"
 
+	"knative.dev/eventing/pkg/apis/feature"
 	"knative.dev/pkg/apis"
 )
 
 func (ep *EventPolicy) Validate(ctx context.Context) *apis.FieldError {
+	// To not allow creation or spec updates of EventPolicy CRs
+	// if the oidc-authentication feature is not enabled
+	if apis.IsInCreate(ctx) || apis.IsInSpec(ctx) {
```

Comment:
Since we are trying to disallow spec updates only here, let's just make it explicit:
```suggestion
	if apis.IsInCreate(ctx) || (apis.IsInUpdate(ctx) && apis.IsInSpec(ctx)) {
```